File: articles/john-backus-fortran-and-the-birth-of-high-level-scientific-programming/cpp/formula_translation.cpp
#include <iostream>
#include <vector>

double formula(double x, double a, double b, double c) {
    return a*x*x + b*x + c;
}

int main() {
    for (double x : std::vector<double>{-2, -1, 0, 1, 2, 3}) {
        std::cout << "x=" << x << ", y=" << formula(x, 2, -3, 1) << "\n";
    }
}

File: articles/john-backus-fortran-and-the-birth-of-high-level-scientific-programming/rust/formula_translation.rs
fn formula(x: f64, a: f64, b: f64, c: f64) -> f64 {
    a * x * x + b * x + c
}

fn main() {
    for x in [-2.0, -1.0, 0.0, 1.0, 2.0, 3.0] {
        println!("x={:.1}, y={:.6}", x, formula(x, 2.0, -3.0, 1.0));
    }
}
